### ISSUE TITLE
Tracker optimization

### ISSUE THREAD

**Issue by mrlika** [OpenWebTorrent/openwebtorrent-www#10]
Hi,

I see that Open Web Torrent project is testing [wt-tracker](https://github.com/Novage/wt-tracker) and I noticed that memory usage in your instance is slightly better than on mine :-)

So I'm just wondering did you do any optimizations in the Node.js environment or in the code?

P. S.
I am going to create a guide on how to run a personal tracker for about 5$/month (OVH or Hetzner) that can handle 40k+ peers and any additional optimizations will be helpful.

**Comment by alxhotel** [comment]
Hi @mrlika,

Great work with `wt-tracker`, is working pretty well.

I haven't done any real optimizations regarding memory. I had to make some changes in my server configuration, such as the number of files open. But I haven't encountered with any memory related problem. Currently, the memory usage of the NodeJS process is around 1% of the total memory. And I also use Nginx to proxy all requests, which uses around 10% of the total memory.

I'm guessing that with 5$/month, the best offer you could get is a 4GB RAM server with one core. In that case, to handle 40k+ peers, the main problem would be the CPU. With only one core it might overload the server.

**Comment by alxhotel** [comment]
@mrlika did you find any other useful tips?

**Comment by mrlika** [comment]
No. Node.js memory usage on high loads is the main concern now to run more peers on low-cost VPS. Working on the C++ implementation. But I don't have a lot of free time for it now...

**Comment by mrlika** [comment]
Record - 49815 peers and after that killed by the kernel because of OOM. I think they were on your tracker this night too.

Using Node.js wt-tracker, 1 CPU core, 2GiB of memory.

Someone is trying to use public trackers in production...

**Comment by mrlika** [comment]
So looks like it is possible to handle up to 50k on such hardware

**Comment by alxhotel** [comment]
Nice! Yeah, I saw a spike in traffic the other day. I didn't see how many peers I got.

Although this implementation seems pretty efficient, I think  we still need a way to scale horizontally.
